51 search page lag

kennedylinzie opened the pull request:
improved searching 
have loading dialog for data
no pause when search is clicked
smooth animating when searching 

added threading to most of the searching pieces

kennedylinzie committed:
lagchanges-test

kennedylinzie committed:
lag improvements

changes:
app\src\main\java\org\apphatchery\gatbreferenceguide\ui\adapters\FAGlobalSearchAdapter.kt
app\src\main\java\org\apphatchery\gatbreferenceguide\ui\fragments\GlobalSearchFragment.kt
app\src\main\res\layout\fragment_global_search_item.xml

test it and let me know how what you think

kennedylinzie committed:
Update FAGlobalSearchAdapter.kt

Wiza-Munthali reviewed (changes requested):
Works well,

### Minor change:
remove the number bellow the progress indicator

### Problem:
There is a lag when going back to the search page if there is no search term

Steps to emulate:
1. Go to search
2. Click any chapter
3. go back to search
4. ISSUE: The loading will take place

https://github.com/AppHatchery/GATBReferenceGuide-Android/assets/42870528/43d3581f-9e0e-4fc1-bce3-2241f291ff7b

kennedylinzie committed:
fix back delay

we create a copy of the data and put less data when its empty so it is not as heavy on the system

kennedylinzie committed:
Merge branch 'master' into 51-search-page-lag

kennedylinzie committed:
Update GlobalSearchFragment.kt

kennedylinzie committed:
search lag hold data in singleton

files changed
app\src\main\java\org\apphatchery\gatbreferenceguide\ui\fragments\GlobalSearchFragment.kt
app\src\main\java\org\apphatchery\gatbreferenceguide\utils\Utils.kt

we now hold the repeated dataset in a asingleton so we dont have to generate it everytime  thus improving the speed then there is no search input

kennedylinzie committed:
cleaning up

Wiza-Munthali reviewed (commented):
The implementation makes sense and looks efficient, where we query all the information on the first occurrence and then use that information through out the search process rather than retrieving the data every time that page is hit. @yago1994 what do you think?

kennedylinzie committed:
Merge branch 'master' into 51-search-page-lag

kennedylinzie committed:
Merge branch 'master' into 51-search-page-lag

Wiza-Munthali commented:
Closed done in another PR
